docs/protocol/extension_trace.md

```diff
@@ -0,0 +1,164 @@
+<!--
+# Copyright (c) 2022, NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+#
+# Redistribution and use in source and binary forms, with or without
+# modification, are permitted provided that the following conditions
+# are met:
+#  * Redistributions of source code must retain the above copyright
+#    notice, this list of conditions and the following disclaimer.
+#  * Redistributions in binary form must reproduce the above copyright
+#    notice, this list of conditions and the following disclaimer in the
+#    documentation and/or other materials provided with the distribution.
+#  * Neither the name of NVIDIA CORPORATION nor the names of its
+#    contributors may be used to endorse or promote products derived
+#    from this software without specific prior written permission.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS ``AS IS'' AND ANY
+# EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+# IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+# PURPOSE ARE DISCLAIMED.  IN NO EVENT SHALL THE COPYRIGHT OWNER OR
+# CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+# EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+# PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
+# PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY
+# OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+# (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+# OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+-->
+
+# Trace Extension
+
+This document describes Triton's trace extension. The trace extension enables
+the client to configure the trace settings during a Triton run. Because this
+extension is supported, Triton reports “trace” in the extensions field of
+its Server Metadata.
+
+## HTTP/REST
+
+In all JSON schemas shown in this document $number, $string, $boolean,
+$object and $array refer to the fundamental JSON types. #optional
+indicates an optional JSON field.
+
+Triton exposes the trace endpoint at the following URL. The client may use
+HTTP GET request to retrieve the current trace setting. A HTTP POST request
+will modify the trace setting, and the endpoint will return the updated trace
+setting on success or an error in the case of failure. 
+
+```
+GET v2/trace
```

Comment:
So all trace settings can be configured at model (/version) level? i.e. it can have say 100 sample rate globally and for model A, we can set 1 sample rate so all requests to model A will be traced?

Reply:
Yes, you should use the commandline values as the initial global defaults that apply to all models/versions. The /v2/trace API can be used to get/set those global defaults. Per-model/version overrides can be set with v2/trace/... API. No need to have per-model controls from command line. So when you are getting the settings for a particular model the order of precedence is model/version settings, model settings, global settings.

Reply:
Do we also want a "clear" API on model and model version to remove the specific settings so user can reset it to use global settings?

Reply:
Yes, that would be good. Not sure if we should just have .../trace/clear endpoint for that or something else.

Reply:
We may add something like when using the POST to update the specific model or model version without providing any parameters, then it will clear the previous specification, although it sounds less clear than having a clear endpoint while I am typing it...

Reply:
I like it better than adding a .../clear endpoint.

Reply:
Pushed a commit to document the clear behavior, and more clarification on specifying model (version) setting without providing all trace setting values